src/app/qgsoptions.cpp

```diff
@@ -1076,6 +1080,44 @@ QgsOptions::QgsOptions( QWidget *parent, Qt::WindowFlags fl, const QList<QgsOpti
     mOptionsStackedWidget->addWidget( page );
   }
 
+#ifdef HAVE_OPENCL
+
+  // Setup OpenCL (GPU) widget
+  mGPUEnableCheckBox->setChecked( QgsOpenClUtils::enabled( ) );
+  if ( QgsOpenClUtils::available( ) )
+  {
+    mGPUEnableCheckBox->setEnabled( true );
+
+    for ( const auto &dev : QgsOpenClUtils::devices( ) )
+    {
+      mOpenClDevicesCombo->addItem( QgsOpenClUtils::deviceInfo( QgsOpenClUtils::Info::Name, dev ), QgsOpenClUtils::deviceId( dev ) );
+    }
+    // Info updater
+    std::function<void( int )> infoUpdater = [ = ]( int )
+    {
+      mGPUInfoTextBrowser->setText( QgsOpenClUtils::deviceDescription( mOpenClDevicesCombo->currentData().toString() ) );
+    };
+    connect( mOpenClDevicesCombo, qgis::overload< int >::of( &QComboBox::currentIndexChanged ), infoUpdater );
+    mOpenClDevicesCombo->setCurrentIndex( mOpenClDevicesCombo->findData( QgsOpenClUtils::deviceId( QgsOpenClUtils::activeDevice() ) ) );
+    infoUpdater( -1 );
+  }
+  else
+  {
+    mGPUEnableCheckBox->setEnabled( false );
+    mGPUInfoTextBrowser->setText( QStringLiteral( "An OpenCL compatible device was not found on your system.<br>"
+                                  "You may need to install additional libraries in order to enable OpenCL.<br>"
+                                  "Please check your logs for further details." ) );
+  }
+
+
+#else
+
+  mOptionsListWidget->removeItemWidget( mOptionsListWidget->findItems( QStringLiteral( "GPU" ), Qt::MatchExactly ).first() );
```

Comment:
That is the visual text, correct? Won't this break on translated builds?

Reply:
It is actually already broken :) Thanks for noting